### ISSUE TITLE
Has this been abandoned?

### ISSUE THREAD

**Issue by Junior1544** [chrippa/livestreamer#1427]
This software looks like it hasn't been updated by the person in control of it in over a year...

Has it been abandoned?  Is there someone else out here that'd start a new git and implement the updates that there are pull's for and keep it going under a new name??

I would myself, except I'm still new to Github and really don't know what I'm doing yet...

Thanks!

**Comment by oyvindln** [comment]
The last bunch of merged pull requests were merged after half a year of inactivity, so it's hard to say.

**Comment by gravyboat** [comment]
I honestly don't know, I've tweeted at the developer asking if he needs donations to keep working on this and so forth but have never received a response.

**Comment by noodle-** [comment]
Judging from the forks graph alone, @FurryFur seems to have improved this tool for a while over [here](https://github.com/FurryFur/livestreamer). Not anymore apparantly though.

**Comment by gravyboat** [comment]
Yeah none of the forks have continued to be maintained.

**Comment by noodle-** [comment]
I'll try my hand at continuing it a bit by simply merging other peoples merge requests into [my repository](https://github.com/noodle-/livestreamer/commits/develop).

I know some python, but I have zero experience with bigger projects like this one, so if you want to help/join/collaborate, feel free to ask & I'll set you as admin or whatever it's called on here.

**Comment by gravyboat** [comment]
@noodle- Yeah add me in. I'm proposing the following remediation plan of action:
1. We go through the develop branch and ensure all the tests pass and determine code coverage (no more merges until that is done and 2 is done).
2. Duplicate and set up the Travis settings to make sure everything is getting tested on a variety of operating systems.
3. Begin the very tedious process of merging in some of these existing PRs and noting where they were merged, if possible retain history for the associated user to try and get them involved. We should also get tests whenever possible for these merges, if the contributor can write them that would be the best.
4. Start looking through the issues, create tags that are available for users in the new repo which we can port over if needed.

My only experience with working on a large project is https://github.com/saltstack/salt/, and I've learned the following from that:

Prioritize proper release naming, no bugs on releases, and proper issue/PR handling over anything else. If we can get more people who are good with Python involved that would be beneficial. My Python is okay but I am by no means a grade A developer.

**Comment by scottbernstein-zz** [comment]
I'm in for testing/validating on Windows OS

**Comment by gravyboat** [comment]
This reminds me that once we have things in a reasonable state we should notify these guys: https://github.com/bastimeyer/livestreamer-twitch-gui

They are actually maintaining their project and deserve to be notified if we're going to get updates in. Actually maybe their community is interested in helping. @bastimeyer I'm adding you onto this conversation. If you have any Python devs you know that love livestreamer we could use some help.

**Comment by Junior1544** [comment]
Maybe even ask them for help with it?

**Comment by gravyboat** [comment]
@Junior1544 I edited my message to mention the primary maintainer.

**Comment by noodle-** [comment]
@Junior1544 My plan was more along the lines of 
1. Look at all the pull requests on here
2. Merge & test whether it works
3. yay

So maybe it's better if you set up the (new) main repository and add me as little helper instead. I don't have time to do much, but I'd be happy to help & learn wherever I can.

**Comment by bastimeyer** [comment]
Thanks for mentioning me here and letting me know...
Sadly, I don't have any python knowledge, all I can do is open an issue on my repo and add a reference to this thread here. Hopefully you'll find a couple of more people that support your fork ideas.

It's a shame that Chrippa seems to have abandoned his project. Personally, I would welcome a fork with all necessary bugfixes and having an assurance that it's being maintained if something breaks and gets fixed in time. We don't have this now and it's quite scary, because it might break any day. Having a working and maintained fork would save me from reimplementing something similar to livestreamer in my application.

The last time something broke regarding viewing Twitch streams, it was fixed within a couple of hours, back in early december 2014. I've heard that there are currently some issues with certain VODs on Twitch that livestreamer is unable to read correctly. That's a problem for the future plans of my application. There are also a couple of player argument parsing bugs in the current version that are around for quite some time.

As I've already said, I'm fine with the idea of a fork, but I don't know how you are planning to do this.
- Are there any licensing issues?
- Would a rename be required? If so, the name of my application wouldn't make sense anymore if I support the fork.
- Will the API/CLI change? Are there any other changes necessary that might break the compatibility of both programs?
- Will the maintainers be able to lead the fork into the right direction or will the community reject or ignore it?

**Comment by Junior1544** [comment]
I have no idea on where to even start, otherwise I would have done it
already...

Sorry.

**Comment by LEChaney** [comment]
I think a new fork with new maintainers might be a good idea at this point. Also sorry to say that my fork is quite a lot of extra code and I didn't write any tests for it. It probably needs tidying up quite a bit. I used a lot of threading, and it's prone to deadlock issues. The mailbox behavior should be changed to work like a golang channel to make it a bit more consistent. When I wrote it I was trying to add code without changing too much existing stuff. A lot of it could probably be rewritten to be a lot simpler.

**Comment by luceatnobis** [comment]
I don't know much about the code of this project but in general I would like to help with this. I would be a lousy maintainer, but if there are open issues in a fork I might take a look at it.

**Comment by LEChaney** [comment]
I've posted some stuff about the issues I currently have with my own fork over at #1311 

I think anyone looking at creating a new fork of this would need to be quite dedicated to it. There are a lot of things to think about like, a new website, checking any licensing issues, setting up continuous integration / deployment of binaries, documentation, making sure there is good test coverage, making sure the tests are actually useful, making sure you don't accept any pull requests into master that would break anything, and continued development of the project. 

I think it would be a good idea for anyone serious about making a new fork to create a Patreon page to support the core maintainers of the package, so that there is something to justify their time input other than "just being a cool project". The fork I made was my first real crack at making some fairly significant changes to an open source project, and it was a lot more work than I thought it would be.

In saying that, you will learn a lot of useful stuff doing this, I'd be tempted to do this myself if I thought I would get good Patreon support and could keep my hours to say 10 hours a week.

**Comment by gravyboat** [comment]
@FurryFur This has been my main concern when I considered taking this over almost a year ago after @chrippa didn't respond even after I asked about donations to the project. It's a big project to take over, and I honestly don't know if the community that uses Livestreamer is even willing to donate to make the amount of work required to maintain it tenable. Even 10 hours a week is a LOT of hours, and for most software developers that's easily a couple grand a month to maintain. I already spend a good chunk of time on one open source project and have received a total of $0 and 3 free meals for my work.

**Comment by Junior1544** [comment]
I don't know about anyone else, but i'd donate to a good developer (or
team) to help get this thing back on track...   I'm trying to use it with
livestream.com and it wont play the stream for more then 10 seconds before
it stops...  I've heard of a way to fix it but I don't know how to put it
in place...

Hopefully someone can get this thing going again, or maybe @chrippa will
allow a few other people here to help him keeping it up and going...

Thanks guys and girls!  this whole discussion is giving me at least some
hope that this project isn't dead!

--James

James Hauge
James Virtual Tours LLC.
http://JamesVirtualTours.com
Hauge.James@gmail.com
808-782-8461

**Comment by oyvindln** [comment]
As a side note, [youtube-dl](https://github.com/rg3/youtube-dl) is another maintained free software project that does have much of the same functionality (though it's more focused on downloading as opposed to streaming.)

**Comment by scottbernstein-zz** [comment]
Hey @junior1544 -- there is an easy way to make livestream.com streams work -- just open the page that's streaming, and use CTRL+U (assuming you're in Google Chrome) then press CTRL+F (for "find") and search for .m3u8.  Take that full URL (including all of the code after it, up until the closing quotes) and that will be the stream URL for livestreamer.  Request it with the hlsvariant:// prefix and you're golden.

**Comment by Junior1544** [comment]
Thanks @scottbernstein !!  But, I just gave that a shot, and am getting a
forbidden error...

the live stream i'm trying to play can be viewed directly on this url:
http://livestream.com/accounts/18265544/events/5004281/player?width=560&height=315&autoPlay=true&mute=false

and the livestream command i tried is:
livestreamer hlsvariant://
api.new.livestream.com/accounts/18265544/events/5004281/broadcasts/132546519.m3u8?dw=100&hdnea=st=1470942579~exp=1470944379~acl=/i/18265544_5004281_lsihzu8nfbyy7wbdl7s_1@193775/*~hmac=e8cf72ad0c8097a5fc3ac9ca5d0cbfebd013edc02bdcf89fe34a628bc2dc92cc
270p_hls --player omxplayer

I also tried putting quotes around the whole url because it was giving me
an error about an invalid command --player without it, with the quotes it
gave me the forbidden command...

Thanks!
--James

James Hauge
James Virtual Tours LLC.
http://JamesVirtualTours.com
Hauge.James@gmail.com
808-782-8461

**Comment by scottbernstein-zz** [comment]
@junior1544

That should work -- with 2 caveats:
1.  You DO need quotes around the full stream (for MS Windows command line at least).
2.  You must issue the command IMMEDIATELY after grabbing that m38 URL.  If you let more than a minute or 2 pass, you need to refresh the page and grab the new M3u8 URL there.

In fact, I just validated your video stream from the URL you used and it worked perfectly.

Most important thing is to refresh the page and grab the m3u8 link JUST before you issue the livestreamer command.

**Comment by Junior1544** [comment]
Thank you @scottbernstein 
I think that there's a rather large issue here....

It's not for a windows computer.  it's for use with Linux, the Raspberry Pi...

And, I did it within a minute...   but, I also need something that'll work for longer than 1 minute, that's why I was hoping that this program was going to start being supported again...

I was told that there was a solution that works, but I don't know how to put it into place, it's a change to a few lines of the code...    it's listed at #1277 

Thanks...

**Comment by scottbernstein-zz** [comment]
yes, I have also confirmed that pull request #1277 will work

**Comment by Junior1544** [comment]
@scottbernstein thanks, the next question would then be, how do I put that in place?  I can't even find the files on my Raspberry Pi to make any changes to it...

**Comment by gravyboat** [comment]
@Junior1544 @scottbernstein Could you guys please stop commenting on that other issue in this thread? It isn't relevant for the current topic.

**Comment by WolfganP** [comment]
Reading the comments above, it seems that no one has the time to properly maintain the overall project, but at the same time there's interest in contribute to it from several people.

Looking at the forks' map (https://github.com/chrippa/livestreamer/network), it seems several devs work regularly on their forks recently (ie @FurryFur @intact @steven7851 ); so maybe a workable strategy would be to integrate those forks as a loosely organized continuation effort? (and invite whoever is submitting PRs in this original project to do it in the continuation as well to add valuable community efforts?)

Just thinking aloud...

**Comment by noodle-** [comment]
@WolfganP Yes, someone who's able & willing will have to step forward to do the honours of hosting the main project/repository though.

**Comment by gravyboat** [comment]
@WolfganP The main problem is whoever is the primary maintainer is going to have a lot of work to do no matter what. We're talking at least a full 40 hours just to get the project back into the barest of shape with merging and reviewing PRs. The work has built up over time unfortunately so now we're in this shitty spot.

**Comment by SpootDev** [comment]
How scary is the codebase on this thing?  Looked at a few files and the py3 compatibility looks a bit hackish.  I was going to use this as an example or fork/refactor in my crazy media stream/transcode suite...but now I'm not so sure. Don't know if I want to walk into 500 bug/issues and 50 pr's.  heh

**Comment by Junior1544** [comment]
Maybe, if anyone is able to get a hold of the current person who 'own's' this git, can get a few more people added as admin's to it and then spread out the work a bit?  I'd do it in a second if I knew what I was doing to begin with...

**Comment by hubbcaps** [comment]
I think someone just needs to take the initiative, call their fork the new "master" and try to reach out to the ones we know maintaining their own forks, ask them to push to your repo, and get the word out. Hopefully we can avoid ego clashes about whose name is on the new master and just run with it.

The longer it stagnates the worse it's going to get.

**Comment by lawli3t** [comment]
I'm definitely willing to help out on some issues / bugs, but taking maintainership of the project is definitely out of question with my current occupation. I'll be following this thread for further updates, I hope something comes out of this. Im using livestreamer daily, and would be glad to be able to help this tool and give something back to it.

**Comment by lawli3t** [comment]
@cdrage

**Comment by cdrage** [comment]
@lawli3t thanks for ping me here, going to cross-post me response from https://github.com/bastimeyer/livestreamer-twitch-gui/issues/289

i'm interesting in devoting some time to this. i have some knowledge of python, and maintenance, however, I'd only be able to devote a few hours a week to this, just enough to maintain it.

anyone want to come up with a decent name / alternative to "livestreamer"?

i propose perhaps "streamlink" as a decent name. get an organization repo up and running, fork it, and open up some PR's to be merged in.

**Comment by cdrage** [comment]
@lawli3t @hubbcaps @Junior1544 @oyvindln @gravyboat @noodle- @scottbernstein @bastimeyer @FurryFur @luceatnobis @WolfganP @SpootDev @hubbcaps 

So I've created https://github.com/streamlink/streamlink which is the initial fork of livestreamer.

Let's have this 100% open with access from ANYONE. Seriously. No more waiting for PR's to get merged or continual pinging of the original maintainer. 

This is the power of Git and open source.

Feel free to submit PR's. I'll be busy for the rest of the day, but I should have a few hours this weekend to devote some time to this. 

Feel free to ping me or email me (it's on my profile) and I'll add you, no matter who you are!

**Comment by gravyboat** [comment]
@cdrage Good stuff, add me to the org please.

**Comment by SEQUOIIA** [comment]
@cdrage Feel free to add me too. I'll gladly help out.

**Comment by LEChaney** [comment]
@cdrage Me too.

**Comment by SpootDev** [comment]
ok, dogpile on @cdrage hehehehehehe   Might as well throw my hat in too.

**Comment by fishscene** [comment]
@cdrage I'll gladly help where I can.

**Comment by cdrage** [comment]
Thanks all! I'm out today, but I'll have time tomorrow to add everyone and
organize the repo.

**Comment by scottbernstein-zz** [comment]
Some things that are broken and need immediate check-ins into this  build (from the most recent updates to the following plugins):

livestream.py

ustream.py

dailymotion.py

periscope.py

Scott

**Comment by bastimeyer** [comment]
@cdrage 
Thanks for stepping up here and initializing this, much appreciated! 👍 
Using a github org and giving the right people write access to the repo is the correct decision, but don't forget that at least one needs to stay responsible for the project, even after months. I've seen other projects like this with multiple authorized contributors going inactive/stale, because noone felt responsible and cared for any open issues/PRs after a while.
Also, the amount of work that needs to be done in order to get this going must not be underestimated. Fixing bugs and merging PRs is just one thing. Packages need to be created for various distros/OSes and maybe also gh-pages set up for the docs and other instructions. Having a plan or a list where everyone can see what needs to be done should help with this.
Thanks a lot! 😃 🎉 👍 😄

**Comment by SpootDev** [comment]
@bastimeye brings up good points.  I have a Jenkins/Proxmox that could possibly be throw into the mix for build/vm, CI, etc.....we'll see.

**Comment by gravyboat** [comment]
@SpootDev I've started several issues over in the other repo to address these points. We should be able to push most of that stuff through Travis to ensure that it stays open and accessible to everyone.

**Comment by javiercantero** [comment]
Really, it's a bad idea to change the name of the project. Are you sure you can't simply maintain a new branch "proposed-updates" or something along the lines and merge all the pending patches over there?

**Comment by cdrage** [comment]
@javiercantero the problem is that since this has been abandoned, we can't maintain / update the packages that are currently on CentOS / Debian / Ubuntu / etc. With a new name (as well as an org instead of a user) we have more control whether this project is abandoned again in the future, etc.

**Comment by cdrage** [comment]
Everyone check your invites, I've added everyone to the org with admin access to http://github.com/streamlink/streamlink if you feel that you need to add another collaborator with push access, etc, feel free to you (should) have the permission to do so.

Keep in mind that the permissions (admin) is very lenient, I'd like us to be as open as possible. Luckily with the power of Git, if someone were to be toxic it's a quick `git reset --hard 1290381023 && git push -f`.

**Comment by javiercantero** [comment]
@cdrage You don't have to maintain the packages, the distro package maintainers will do it. Believe me, it's going to be more difficult to add a new package to Debian in time for the next stable release (the freeze is in 3 months and the package should be accepted and be ready before that) that to convince the current livestreamer Debian maintainter to add some patches downstream (this applies not only to Debian but also to any derivative including Ubuntu since the Debian package is the basis for the rest).

**Comment by LoneFox78** [comment]
While others are busy talking about doing things and therefore not actually doing anything, I went through all the pull requests and merged most of them into the [master branch of my repo](https://github.com/LoneFox78/livestreamer). Included are:
- #1203 Dplay + removal of defunct plugin (UNTESTED, all streams seem to be geo restricted)
- #1234 New plugin for bilibili.com
- #1240 Support afreecatv.jp domain (Appears to not work... and why there are two separate plugins for afreeca?)
- #1241 New plugin for panda.tv
- #1243 New plugin for sportschau.de (UNTESTED, no live streams available, does not work for non-livestream videos on the site)
- #1251 Support twitch host mode
- #1252 itv.com fixes (UNTESTED, all streams seem to be geo restricted)
- #1253 New plugin for mediaklikk.hu
- #1257 New plugin for tvthek.orf.at
- #1277 livestream.com fix
- #1289 tga plugin fix/update
- #1291 Fix for youtube
- #1329 New plugin for rtl.nl
- #1332 Fix for picarto.tv
- #1333 New plugin for openrec.tv
- #1352 Fix for tv.nrk.no
- #1359 Update twitch oauth redirect URL
- #1375 Fix for svtplay
- #1376 Fix for vaughnlive
- #1378 New plugin for vidio.com
- #1383 Fix for afreecatv
- #1397 Fix for twitch
- #1401 Another fix for afreecatv
- #1414 Fix for douyu.com
- #1429 Fix some twitch stream names
- #1439 Documentation fix
- #1450 Fix livecoding.tv vods
- #1461 Fix for goodgame.ru
- #1469 Another fix for picarto.tv
- #1476 New plugin for viafree.dk
- #1483 twitch client ID patch

and these didn't get in:
- #233  Abandoned back in 2013, don't know why
- #1024 I'm not sure what this is good for...
- #1105 Python 3.6 fix, we'll think about it once python 3.6 is released.
- #1160 goodgame.ru fix superseded by #1461
- #1167 VLC specific, can't test without VLC, and it probably breaks other players that don't support --meta-title argument.
- #1197, #1227, #1278 ustream.tv fixes conflict with each other. Some comments say that the whole plugin should be rewritten. Need someone who actually knows this stuff to sort it out.
- #1200 Documentation change, I'm not sure if this belongs here...
- #1204, #1210, #1298, #1311 I didn't want to add any big new features to this. Let's fix all broken things first.
- #1206, #1214, #1274, #1302 streamup.com, castamp.com, immono.com, and periscope.tv. These seem to require some sort of mobile app to use (no stream list on the websites), meaning that I don't have any means of testing them. Personally, I think that nothing of value is lost if this hipster stuff is left out. If someone else feels otherwise, go ahead and include them.
- #1238 Looks like some sort of configuration error on reporter's computer...
- #1242, #1364 douyu fixes superseded by #1414
- #1267 BS on wheels? For me, the Right Way™ to release software is a source tarball. Someone who actually uses the binary nonsense should handle this one.
- #1269 According to comments, this fix has problems and the videos are geo restricted, so I cannot test.
- #1280 Another bilibili.com plugin, duplicate of #1234
- #1386 Documentation change that may not be true anymore. Should wait until the new upstream is well established.

Now I'm trying to find out how to upload a release tarball here, so that all the Technologically Impaired Ducks using the program can get a working version without needing to deal with git...

Also, hasn't anyone noticed that the streamlink repository is cloned from master rather than develop branch and therefore is missing all commits between 1.12.2 release and the last merge at March of this year?

**Comment by sondr3** [comment]
> Really, it's a bad idea to change the name of the project. Are you sure you can't simply maintain a new branch "proposed-updates" or something along the lines and merge all the pending patches over there?

I disagree, when the community agrees that changes need to happen I think the best way is to completely fork it and start a new project, but honor the legacy from which it came. LibreOffice is a good example in my mind.

**Comment by javiercantero** [comment]
@sondr3 That only applies when a community splits in two different ways. That's not what is happening here.

**Comment by LoneFox78** [comment]
A tarball + Windows installation instructions are now available [here](https://github.com/LoneFox78/livestreamer/releases/tag/1.14.0-rc1). Again, it's meant to be a temporary solution until the streamlink folks have a better release available.

**Comment by javiercantero** [comment]
@LoneFox78 Don't bother with any PR < 1192 (the last one merged by chrippa)

**Comment by cdrage** [comment]
Hey @LoneFox78 thank you so much for all the work you've done for your fork / branch.

IMO, we should still fork / split as unfortunately we can't get anything merged into livestreamer as @chrippa has abandoned it. 

A good start would be to merge your branch / fork into streamlink. 

I've created the PR https://github.com/streamlink/streamlink/pull/9 so we can have a great starting point / code-base for everything.

**Comment by alphapapa** [comment]
I'm just a bystander, but my two cents: While it would be ideal to receive stewardship of the project from @chrippa, in the absence of that, forking and renamin g is the only viable path forward.  If this, the canonical repo for livestreamer, is to remain stuck, there must be a new name for the fork to distinguish it.

**Comment by javiercantero** [comment]
#1495 is my own selection of urgent fixes. It's based on the list provided by @LoneFox78 above but shorter since it doesn't include new code, new features or new plugins, only fixes for existing plugings.

My goal is to try to convince the Debian maintainers to include the patch in the livestreamer package for the next release of Debian (9.0) before it is too late. This also should affect any derivative of Debian (including Ubuntu and its derivatives).

Any testing of the patched plugins is greatly appreciated.

**Comment by gravyboat** [comment]
@javiercantero How long do we have for the Debian release? We've got a working release of streamlink that we might be able to push if we can package it up.

**Comment by javiercantero** [comment]
@gravyboat The hard deadline for the "freeze" of Debian Stretch is January 5, 2017 but I recommend you to start ASAP (since I don't know if the inclusion of new packages is welcome so close to the freeze date).

EDIT: simply to confirm that the deadline for new packages is [January 5th 2017](https://nthykier.wordpress.com/2016/10/01/stretch-transition-freeze-in-a-month/)

**Comment by gravyboat** [comment]
The joy of slow moving projects, 3 months to go is 'close' to the freeze date.

**Comment by alphapapa** [comment]
@javiercantero Thanks for your effort to get the patches into Debian.  As a Debian user, I appreciate it!

**Comment by javiercantero** [comment]
@gravyboat The whole process is (at least) two years, so 3 months is really close :)

**Comment by maweki** [comment]
I am the author of the twitchlive extension for gnome shell (https://github.com/maweki/twitchlive-extension) and for now I will be pointing people to streamlink

**Comment by grazzolini** [comment]
@LoneFox78
I noticed this project was abandoned for a while now. But I come back to see how was my pr #1204 going. I've been using it ever since I coded. I disagree it's a big change, as you've put it. I changed very few lines. And this option is great for streaming and recording at the same time. Best of both worlds. Nice to see things moving, though.

**Comment by alphapapa** [comment]
@grazzolini That looks like a very nice addition!  I agree that it seems like few lines for adding such a feature.  Well done.

**Comment by scottbernstein-zz** [comment]
@giancarlo -- please issue the same pull request to the Streamlink project which has taken over development of livestreamer as an active project:

https://github.com/streamlink/streamlink

[https://avatars0.githubusercontent.com/u/22243823?v=3&s=400]https://github.com/streamlink/streamlink

GitHub - streamlink/streamlink: CLI for extracting streams ...https://github.com/streamlink/streamlink
github.com
streamlink - CLI for extracting streams from various websites to video player of your choosing [fork of livestreamer]

It started off with the same code base and has only recently started making some changes

**Comment by lastmetroid** [comment]
I tested streamlink and it's just the most user unfriendly BS ever (for Windows). There is no prebuild Windows build just some nonsense GUI which downloads and (compiles) it together, first nonsense BS. Then there is no config file to find like it was in livestreamer, to define which player should be used, VLC ect. Then how to use it ? Just replaced livestreamer.exe with the streamlink.exe doesn't work ... it seems not to support the same commands. This is unbelievable.

**Comment by hubbcaps** [comment]
@makedir Perhaps you should make a new issue in the Streamlink repo with your issue with a coherent statement of your problems rather than necro posting in this thread on an abandoned project. You'll be much more likely to get help over there.

**Comment by fishscene** [comment]
Yes, Please raise an issue over at [Streamlink](https://github.com/streamlink/streamlink/issues)

Livestreamer was forked several months ago and has understandably changed a bit to meet changes in API's and to meet the needs of those using it. But let's discuss your concerns over at Streamlink. :)

**Comment by scottbernstein-zz** [comment]
For me (on Windows), Streamlink has been exactly a drop in replacement.  Sorry it doesn't work for you....

On Apr 7, 2017, at 11:31 AM, fishscene <notifications@github.com<mailto:notifications@github.com>> wrote:


Yes, Please raise an issue over at Streamlink<https://github.com/streamlink/streamlink/issues>

Livestreamer was forked several months ago and has understandably changed a bit to meet changes in API's and to meet the needs of those using it. But let's discuss your concerns over at Streamlink. :)

**Comment by rbong** [comment]
@LoneFox78 Castamp is not mobile only, it just does not have public streams. Also, it's not untestable, I even have an example of testing it here: https://github.com/rbong/livestreamer-nightly/tree/castamp

However it has been several years since I used this software so I will likely not be porting it over to your fork, so if someone wants castamp support unfortunately they will have to add it themselves.
